### DISCUSSION TITLE
RPC services connecting only to loopback ip?

### DISCUSSION THREAD

**Discussion by skotze** [Chia-Network/chia-blockchain#1790] [Q&A]
Hi Everyone

In Windows it would appear that the RPC services are only listening on the local (loopback) ip(s). (see the screenshot below).
This makes it tricksy to do RPC calls from other hosts. (for example building a script running on another host that does RPC calls)

The docs do state that: "The rpc ports should not be exposed to the internet." but as it stands I can't connect to rpc services running on my lan from other hosts due to this binding.

I can obviously do RPCs when I run the call from the same host.

The Non RPC ports (8444 etc) are binding to the network address and I can access them from other hosts (also expected)

Is this intended or am I missing something ?

![image](https://user-images.githubusercontent.com/5038865/114270525-c7f1ee80-9a04-11eb-89a7-abad83badd7c.png)

**Comment by hoffmang9** [comment]
We intended this so that only people who understood the consequences would change it. You can change from localhost to the correct IP address in config.yaml.

**Reply by skotze** [reply]
Excellent, thank you!

**Reply by skotze** [reply]
Actually, Looking at config.yaml, I can see where the rpc ports are set for each service, and I can see how the peer host:ports are set.

However, I'm not able to find the configuration item for setting the ip/hostname that a particular service will bind to.


As an example: (subsection of config.yaml)
full_node:
 ...
  farmer_peer:
    host: localhost
    port: 8447
 ...
 rpc_port: 8555
 port: 8444
...

Nowhere in the /full_node element do I see an element that allows me to set the ip/hostname that the full_node service will bind to.

Is there documentation for the config.yaml that I've just not been able to find?

**Reply by hoffmang9** [reply]
`host: localhost` is what you need to change.

**Reply by skotze** [reply]
`/full_node/host?`

that "path" doesn't exist in the generated yaml.

I've manually added that config item (I saw that `/introducer/host` existed) earlier and have tried it with my `hostname/0.0.0.0/and my actual device ip`, with no changes in the resultant service bindings after restarting the services

Thanks for your help though, I'll play with this more later this week.
